rsocket-core/src/main/java/io/rsocket/plugins/InterceptorRegistry.java

```diff
@@ -30,16 +31,54 @@
  * </ul>
  */
 public class InterceptorRegistry {
-  private List<RSocketInterceptor> requesterInteceptors = new ArrayList<>();
-  private List<RSocketInterceptor> responderInterceptors = new ArrayList<>();
+  private List<Supplier<? extends RequestInterceptor>> requesterRequestInterceptors =
+      new ArrayList<>();
+  private List<Supplier<? extends RequestInterceptor>> responderRequestInterceptors =
+      new ArrayList<>();
+  private List<RSocketInterceptor> requesterRSocketInterceptors = new ArrayList<>();
+  private List<RSocketInterceptor> responderRSocketInterceptors = new ArrayList<>();
   private List<SocketAcceptorInterceptor> socketAcceptorInterceptors = new ArrayList<>();
   private List<DuplexConnectionInterceptor> connectionInterceptors = new ArrayList<>();
 
+  /** Add an {@link RequestInterceptor} that will hook into Requester RSocket requests' phases. */
```

Comment:
It would be mention in the Javadoc to explain that unlike an `RSocketInterceptor` which is a singleton, here a `RequestInterceptor` is per `RSocket` and the `Supplier` is used to create a new instance on each new connection. 

The same comment applies to the other registration methods as well and I think a similar comment can be added in the type level Javadoc of `RequestInterceptor` that it's meant to be used with a separate instance per `RSocket` although technically nothing prevents always supplying the same instance too if you want a singleton.

Another way to achieve this would be to take a `Function<RSocket, ? extends RequestInterceptor>` which would make it more clear and also allow potentially a singleton to be used.

Reply:
fixed